contrib/dracut/90zfs/zfs-env-bootfs.service.in

```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Set BOOTFS environment for dracut
+Documentation=man:zfs(8)
+DefaultDependencies=no
+After=zfs-import.target
+Requires=zfs-import.target
```

Comment:
This is not correct and will break systems.  This should be ordered `After=zfs-import-*.service` and `Before=zfs-import.target`.

Reply:
It is imperative that the `BOOTFS` variable be set *before* reaching `zfs-import.target` because other parts of the initial RAM disk (and quite likely third-party programs too) have come to depend on that assumption.

Reply:
No. You need to import a pool first before reading the bootfs parameter. This was initially a ExecStartPost, so After is correct.

Reply:
Ah, wait... yes, you are right.. everything should be inside the import target

Reply:
@Rudd-O I looked at this again. `zfs-env-bootfs` is pulled in by the generated `zfs-enhancement.conf` [here](https://github.com/zfsonlinux/zfs/pull/8510/files/c1fb1e7b1493e935078c3ea0fd246e8afdfa25db#diff-6809a8a118df0d43b694eb6227b25cc7R47)

`zfs-enhancement.conf` loads after `zfs-import.target` and that is the only part of the initramfs that depends on `BOOTFS`.

While I believe that this would *not* break anything, I move `zfs-env-bootfs.service` to `zfs-import.target`, nevertheless, because this way we make sure we do not change the previous behaviour.